### PR TITLE
Handle all console candidate snippets in docs outside of the reference

### PR DESCRIPTION
**Pull request by nik9000**
Handles all the console candidate snippets in the docs outside of the reference by:
- Remove most of the need for `// NOTCONSOLE` by checking if the shell contain `curl`.
- Add `// COSNSOLE` to check-running and some plugin docs
- Use `{version}` and `{lucene_version}` for versions in the snippets
- Mark groovy snippets in the groovy client as `groovy` rather than `js`

### CODE REVIEW

**Comment by nik9000** [comment]
This drops the number of unhandled console candidates to about 810.

File: docs/plugins/discovery-azure-classic.asciidoc [hunk 1]
```diff
@@ -359,7 +357,7 @@ ssh azure-elasticsearch-cluster.cloudapp.net
 
 Once connected, install Elasticsearch:
 
-[source,sh]
+["source","js",subs="attributes,callouts"]
```
Comment (nik9000) [inline docs/plugins/discovery-azure-classic.asciidoc]: Ah! This should be `["source","sh",subs="attributes,callouts"]` I think.

File: docs/plugins/discovery-azure-classic.asciidoc [hunk 2]
```diff
@@ -372,36 +370,43 @@ sudo apt-get install oracle-java8-installer
...
 
 # Prepare Elasticsearch installation
-sudo dpkg -i elasticsearch-2.0.0.deb
+sudo dpkg -i elasticsearch-{version}.deb
 ----
 
 Check that elasticsearch is running:
 
-[source,sh]
+[source,jso]
```
Comment (nik9000) [inline docs/plugins/discovery-azure-classic.asciidoc]: s/jso/js/

**Comment by nik9000** [comment]
@clintongormley pointed out to me that the code that extracts the snippets probably doesn't support the `["source","js",subs="attributes,callouts"]` form. Which'd be bad for this. I'll run that down.

**Comment by nik9000** [comment]
Removing `review` for now until I fix the busted language detection. I'll add it back soon.

**Comment by nik9000** [comment]
@clintongormley this is ready for another round whenever you get a chance.

**Comment by nik9000** [comment]
@clintongormley, sorry to bother you again, but can you look at this? I'd like to work on some follow ups for things like failing the build if we add more non-console snippets.

File: docs/reference/index.asciidoc [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
-:version:       5.0.0-alpha5
-:major-version: 5.x
-:branch:        master
-:jdk:           1.8.0_73
-:defguide:      https://www.elastic.co/guide/en/elasticsearch/guide/master
-:plugins:       https://www.elastic.co/guide/en/elasticsearch/plugins/master
-:javaclient:    https://www.elastic.co/guide/en/elasticsearch/client/java-api/master/
-:issue:         https://github.com/elastic/elasticsearch/issues/
-:pull:          https://github.com/elastic/elasticsearch/pull/
+:version:        5.0.0-alpha6
```
Comment (clintongormley) [inline docs/reference/index.asciidoc]: Let's stick with what has already been released here, ie `5.0.0-alpha5`

File: docs/plugins/index.asciidoc [hunk 4]
```diff
@@ -1,7 +1,9 @@
 = Elasticsearch Plugins and Integrations
 
-:ref: https://www.elastic.co/guide/en/elasticsearch/reference/master
-:guide: https://www.elastic.co/guide
+:ref:            https://www.elastic.co/guide/en/elasticsearch/reference/master
+:guide:          https://www.elastic.co/guide
+:version:        5.0.0-alpha6
```
Comment (clintongormley) [inline docs/plugins/index.asciidoc]: Let's stick with what has already been released here, ie `5.0.0-alpha5`

**Comment by clintongormley** [comment]
Nice work @nik9000 - thanks!. Couple of minor comments otherwise LGTM

**Pull request merged**